### PR TITLE
Add input types support in TF.optimizers

### PR DESCRIPTION
**Pull request by CatherineSue**
TF.optimizers should accept any input types, such as dict, tuple, list.

Refer to: #111

### CODE REVIEW

File: garage/tf/misc/tensor_utils.py [hunk 1]
```diff
@@ -123,7 +123,14 @@ def pad_tensor_dict(tensor_dict, max_len):
     return ret
 
 
-class enclosing_scope(object):
+def normalized_inputs(inputs, instance=()):
+    if isinstance(inputs, dict):
+        return type(instance)(inputs.values())
```
Comment (ryanjulian) [inline garage/tf/misc/tensor_utils.py]: values() is not guaranteed to have a stable order. you may need to sort by keys first.

File: garage/tf/misc/tensor_utils.py [hunk 2]
```diff
@@ -123,7 +123,14 @@ def pad_tensor_dict(tensor_dict, max_len):
     return ret
 
 
-class enclosing_scope(object):
+def normalized_inputs(inputs, instance=()):
```
Comment (ryanjulian) [inline garage/tf/misc/tensor_utils.py]: can you explain the purpose of this function?

Comment (CatherineSue) [reply garage/tf/misc/tensor_utils.py]: Transform the `inputs` to the same type as `instance`. I found many functions in tf.optimizers have `extra_inputs`,  so calling `inputs = normalized_inputs(inputs, extra_inputs)` transforms two into the same type.

Comment (CatherineSue) [reply garage/tf/misc/tensor_utils.py]: Transform the `inputs` to the same type as `instance`. I found many functions in tf.optimizers have `extra_inputs` but not in the same type, some have type list, some tuple. Calling `inputs = normalized_inputs(inputs, extra_inputs)` transforms two into the same type.

**Pull request closed without merging**